### ISSUE TITLE
SyntaxError: 'super' keyword unexpected here

### ISSUE THREAD

**Issue by doowb** [redwoodjs/redwood#5587]
## Overview

The above error is occurring during api-side transpiling when I extend a class with a method called `.find` and I call `super.find` inside the child class method.

This happens with both, `yarn rw dev` and `yarn rw test api`.

## Input

```js
class MyParentClass {
  find() {
    return 'foo';
  }
}

class MyChildClass extends MyParentClass {
  find() {
    return super.find();
  }
}
```

## Output

The following is found in `.redwood/prebuild/api/src/lib/my-class.js`:

```js
import _findInstanceProperty from "@babel/runtime-corejs3/core-js/instance/find";
class MyParentClass {
  find() {
    return 'foo';
  }

}

class MyChildClass extends MyParentClass {
  find() {
    var _context;

    return _findInstanceProperty(_context = super).call(_context);
  }

}
```

The following error is shown in the terminal:

```shell
    ~/my-project/api/src/lib/my-class.js:91
        return (0, _find.default)(_context = super).call(_context);
                                             ^^^^^

    SyntaxError: 'super' keyword unexpected here
```

## Additional Information

This seems to be caused by a combination of babel and core-js attempting to polyfill the `.find` method on my class.
This is only happening when calling `super.find`. As a workaround, the following code transpiles and works correctly:

```js
class MyParentClass {
  find() {
    return this.internalFind();
  }

  internalFind() {
    return 'foo';
  }
}

class MyChildClass extends MyParentClass {
  find() {
    return super.internalFind();
  }
}
```

**Comment by simoncrypta** [comment]
I'm not an expert with babel, but I don't think the combination of babel and core-js is the cause of this. It supposes to be [“tightly integrated”](https://github.com/zloirock/core-js/blob/master/docs/2019-03-19-core-js-3-babel-and-a-look-into-the-future.md#babel). We can probably change some config to solve this, but your workaround seem to be a good alternative.

You can find our babel config here : https://github.com/redwoodjs/redwood/blob/869eef72fc158904f1d099873cc499e5a6c58eee/babel.config.js#L26

Feel free to open a PR if you want.

**Comment by zloirock** [comment]
This issue can happen if you forget to exclude `core-js` from transpiling - it causes circular dependencies.

**Comment by zloirock** [comment]
@dac09 hmm... What is `@babel/plugin-transform-runtime` if not transpiling? -) Why do you think that your test case should not contain `'@babel/runtime-corejs3/core-js/instance/find'`? Babel is not so smart to understand that it's not inherited, for example, from `Array`.

**Comment by dac09** [comment]
> What is @babel/plugin-transform-runtime if not transpiling?

Haha I guess I meant transpiling in the sense that, transforming latest ES syntax to ES5. Not sure how we can exclude core-js if we're only using the transform-runtime plugin to add polyfills. 

> Babel is not so smart to understand that it's not inherited, for example, from Array.

Yeah thought it might be the case. The reason the test case shouldn't contain it, is because we defined find in the super class - and not using `Array.find`. Is the only solution here to suggest not using "find" then?

What's interesting to note is that using `babel-plugin-polyfill-corejs3` instead of transform runtime doesn't seem to create the issue, but it also doesn't do the polyfill in my test case if I add 

```
const myArray = [];
myArray.find('bazinga');
```

**Comment by zloirock** [comment]
> Is the only solution here to suggest not using a member function called "find" then?

Yes, now it's the only solution if you use `@babel/preset-env` / `@babel/transform-runtime` / `babel-transform-corejs3` (except for a few dirty hacks).

**Comment by Josh-Walker-GM** [comment]
Hey all! I just tried to reproduce this issue on the latest stable version for redwood (6.3.1). I couldn't get the error to show up. The contents in the `.redwood/prebuild` didn't contain the polyfill culprit.

I can't say for sure when this would have been fixed or pinpoint a PR that did so. I'm also not super experienced in this area of our code so please keep that in mind.

I'll likely close this issue after a little bit more time if there is no more discussion here to highlight this is still an issue people are facing. In the meantime I'll try to track down when this could have been fixed.
